storage/blockchain/src/service/types.rs

```diff
@@ -3,28 +3,18 @@
 //! Only used internally for our `tower::Service` impls.
 
 //---------------------------------------------------------------------------------------------------- Use
-use futures::channel::oneshot::Sender;
-
 use cuprate_database::RuntimeError;
-use cuprate_helper::asynch::InfallibleOneshotReceiver;
-use cuprate_types::blockchain::BCResponse;
+use cuprate_database_service::{DatabaseReadService, DatabaseWriteHandle};
+use cuprate_types::blockchain::{BCReadRequest, BCResponse, BCWriteRequest};
 
 //---------------------------------------------------------------------------------------------------- Types
 /// The actual type of the response.
 ///
 /// Either our [`BCResponse`], or a database error occurred.
 pub(super) type ResponseResult = Result<BCResponse, RuntimeError>;
 
-/// The `Receiver` channel that receives the read response.
-///
-/// This is owned by the caller (the reader/writer thread)
-/// who `.await`'s for the response.
-///
-/// The channel itself should never fail,
-/// but the actual database operation might.
-pub(super) type ResponseReceiver = InfallibleOneshotReceiver<ResponseResult>;
+/// The blockchain database write service.
+pub type BCWriteHandle = DatabaseWriteHandle<BCWriteRequest, BCResponse>;
 
-/// The `Sender` channel for the response.
-///
-/// The database reader/writer thread uses this to send the database result to the caller.
-pub(super) type ResponseSender = Sender<ResponseResult>;
+/// The blockchain database read service.
+pub type BCReadHandle = DatabaseReadService<BCReadRequest, BCResponse>;
```

Comment:
`Blockchain` not `BC`.

Reply:
I really can't see anyone confusing this when the docs literally say `blockchain`. In a previous PR the request & response enums were prefixed with `BC` do you want me to change them as well?

Reply:
Many of the abbreviations in the codebase are weird because they aren't commonly used, if at all, e.g. blockchain is not usually abbreviated as `BC`. It also goes against `CamelCase`.

If you think `BlockchainReadHandle` is unbearable (I prefer this by a lot), it could be `BcReadHandle`.

> do you want me to change them as well?

It's ultimately optional, but yes. I'm just saying you deviate from `std` and pretty much every single crate in existence when you do this.

Reply:
> Many of the abbreviations in the codebase are weird because they aren't commonly used, if at all, e.g. blockchain is not usually abbreviated as BC

bc is used throughout monero core

> It also goes against CamelCase

Yeah I'll change this.

> If you think BlockchainReadHandle is unbearable

It wouldn't be so bad here but interacting with the DB through the service interface is already verbose, and I would rather be consistent here so I think making them both `Bc` is best.

Reply:
This wasn't changed.

Reply:
> bc is used throughout monero core

Yes, but I think language convention has higher priority here, `BC` is not a widely used abbreviation.

- https://reth.rs/docs/?search=blockchain
- https://doc-internal.zebra.zfnd.org/zebrad/?search=blockchain

I'm okay with `Bc` though.

Reply:
I'll change this to `Blockchain`.

For the future I would appreciate, with the requested change, a small explanation, even if it's just "I prefer X here".